src/Sylius/Bundle/ApiBundle/Doctrine/QueryItemExtension/OrderMethodsItemExtension.php

```diff
@@ -86,7 +86,7 @@ private function applyForVisitor(QueryBuilder $queryBuilder, string $rootAlias,
             ->andWhere($queryBuilder->expr()->orX('user IS NULL', sprintf('%s.customer IS NULL', $rootAlias)))
         ;
 
-        if ($operationName !== 'shop_select_payment_method') {
+        if ($operationName !== 'shop_select_payment_method' && $operationName !== 'shop_guest_change_payment_method') {
```

Comment:
We should think about some better way to configure this, maybe using sth like `supports` method in the query extensions?